openmdao/recorders/sqlite_reader.py

```diff
@@ -648,7 +648,12 @@ def get_case(self, case_id, recurse=False):
 
         for table in tables:
             case = table.get_case(case_id)
-            if case:
+            if case and case.outputs is not None:
+                for key, value in case.outputs.items():
+                    if isinstance(case[key], list):
+                        case.outputs[key] = np.asarray(value)
```

Comment:
This is the wrong place to fix this... 

The current logic is here:
https://github.com/OpenMDAO/OpenMDAO/blob/master/openmdao/utils/record_util.py#L192
It only converts to an array if there is no shape associated with the value...  allowing for an actual list to be serialized and deserialized.

I think a more correct fix is to add the shape to the metadata in `add_discrete_input` and `add_discrete_output`  (in both component.py and indepvarcomp.py) if the type of the discrete is an array.